contracts/rust/src/light_client.rs

```diff
@@ -0,0 +1,405 @@
+//! Helpers and test mocks for Light Client logic
+
+use std::collections::HashMap;
+
+use ark_ed_on_bn254::EdwardsConfig;
+use ark_std::rand::rngs::StdRng;
+use ark_std::rand::{CryptoRng, Rng, RngCore};
+use ark_std::str::FromStr;
+use ark_std::UniformRand;
+use diff_test_bn254::{field_to_u256, u256_to_field};
+use ethers::{abi, utils};
+use ethers::{
+    abi::{AbiDecode, Token},
+    prelude::{AbiError, EthAbiCodec, EthAbiType},
+    types::{H256, U256},
+};
+use hotshot_stake_table::vec_based::StakeTable;
+use hotshot_state_prover::circuit::PublicInput;
+use hotshot_state_prover::Proof;
+use hotshot_types::traits::stake_table::StakeTableScheme;
+use hotshot_types::{light_client::LightClientState, traits::stake_table::SnapshotVersion};
+use jf_primitives::pcs::prelude::UnivariateUniversalParams;
+use jf_primitives::signatures::schnorr::Signature;
+use jf_primitives::signatures::{
+    bls_over_bn254::{BLSOverBN254CurveSignatureScheme, VerKey as BLSVerKey},
+    SchnorrSignatureScheme, SignatureScheme,
+};
+use jf_utils::test_rng;
+
+type F = ark_ed_on_bn254::Fq;
+type SchnorrVerKey = jf_primitives::signatures::schnorr::VerKey<EdwardsConfig>;
+type SchnorrSignKey = jf_primitives::signatures::schnorr::SignKey<ark_ed_on_bn254::Fr>;
+
+/// Stake table capacity used for testing
+pub const STAKE_TABLE_CAPACITY: usize = 10;
+
+/// Mock for system parameter of `MockLedger`
+pub struct MockSystemParam {
+    /// max capcity of stake table
+    st_cap: usize,
+    /// number of blocks per epoch
+    blk_per_epoch: u32,
+}
+
+impl MockSystemParam {
+    /// Init the system parameters (some fixed, some adjustable)
+    pub fn init(blk_per_epoch: u32) -> Self {
+        Self {
+            st_cap: STAKE_TABLE_CAPACITY,
+            blk_per_epoch,
+        }
+    }
+}
+
+/// Mock of hotshot ledger for testing LightClient.sol functionalities only
+/// its logic is completely divergent from a real light client or HotShot
```

Comment:
```suggestion
/// Its logic is completely divergent from a real light client or HotShot.
```

Reply:
fixed in [6171539](https://github.com/EspressoSystems/espresso-sequencer/pull/854/commits/61715390c68c49dcf05f32a2505fc782957d9628)